scripts/consume_sitemaps.py

```diff
@@ -79,6 +86,26 @@ def thing_fetcher_for_url(thing_url: str, rsession) -> ThingFetcher:
     return thing_fetcher
 
 
+def construct_thing_futures(
+    thing_futures: list,
+    sitemap_file_iterator: Iterator,
+    rsession: requests.sessions,
+    thing_executor: ThreadPoolExecutor,
+    sitemap_file_fetcher: SitemapFileFetcher,
+) -> bool:
+    fetched_all_things_for_current_sitemap_file = False
+    while len(thing_futures) < CONCURRENT_DOWNLOADS:
+        try:
+            thing_fetcher = thing_fetcher_for_url(next(sitemap_file_iterator), rsession)
+            # Add the download job to the queue and remember it
+            thing_future = thing_executor.submit(thing_fetcher.fetch_thing)
+            thing_futures.append(thing_future)
+        except StopIteration:
+            logging.info(f"Finished all the requests for f{sitemap_file_fetcher.url}")
+            fetched_all_things_for_current_sitemap_file = True
+    return fetched_all_things_for_current_sitemap_file
```

Comment:
Refactor the future construction to a method as we now call this in multiple places.